plugin/trino-exchange-filesystem/src/test/java/io/trino/plugin/exchange/filesystem/containers/MinioStorage.java

```diff
@@ -33,26 +38,57 @@ public class MinioStorage
 {
     public static final String ACCESS_KEY = "accesskey";
     public static final String SECRET_KEY = "secretkey";
+    public static final String KMS_KEY_ID = "kms_key_id";
 
     private final String bucketName;
     private final Network network;
     private final Minio minio;
+    private final Optional<Kes> kes;
 
     public MinioStorage(String bucketName)
+    {
+        this(bucketName, false);
+    }
+
+    public MinioStorage(String bucketName, boolean withKes)
     {
         this.bucketName = requireNonNull(bucketName, "bucketName is null");
         this.network = newNetwork();
-        this.minio = Minio.builder()
+        this.kes = withKes ? Optional.of(Kes.builder().withNetwork(network).build()) : Optional.empty();
+        this.minio = buildMinio(kes);
+    }
+
+    private Minio buildMinio(Optional<Kes> kes)
+    {
+        Minio.Builder minioBuilder = Minio.builder()
                 .withNetwork(network)
                 .withEnvVars(ImmutableMap.<String, String>builder()
                         .put("MINIO_ACCESS_KEY", ACCESS_KEY)
                         .put("MINIO_SECRET_KEY", SECRET_KEY)
+                        .buildOrThrow());
+
+        kes.ifPresent(actualKes -> minioBuilder
+                .withEnvVars(ImmutableMap.<String, String>builder()
+                        .put("MINIO_ACCESS_KEY", ACCESS_KEY)
+                        .put("MINIO_SECRET_KEY", SECRET_KEY)
```

Comment:
Do we need this to be set once again ?

Reply:
The first one for Minio, the second for KMS